src/uct/base/uct_iface.c

```diff
@@ -66,10 +69,48 @@ ucs_status_t uct_iface_set_am_handler(uct_iface_h tl_iface, uint8_t id,
 
     if (cb == NULL) {
         uct_iface_set_stub_am_handler(iface, id);
+        return UCS_OK;
+    }
+
+    /* flags != 0 */
+    if (flags == 0) {
+        ucs_debug("am cb flags must not be empty");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    status = uct_iface_query(tl_iface, &attr);
```

Comment:
maybe we don't want to query iface every time active message is registered? anyways transport would fail if cannot invoke SYNC callback from another thread